python/ql/src/semmle/python/dataflow/Implementation.qll

```diff
@@ -0,0 +1,849 @@
+import python
+import semmle.python.security.TaintTracking
+private import semmle.python.objects.ObjectInternal
+private import semmle.python.pointsto.Filters as Filters
+import semmle.python.dataflow.Legacy
+
```

Comment:
I feel like a short worked example would do wonders here. Especially showing how/when the attribute path and parameter contexts change would help a lot with the understanding. (This can of course be gleaned by going through `unprunedStep` and its helper predicates, but it's useful to have an intuition to begin with.)

Reply:
I've added a fairly extensive example to the tests and put a link to it here.